File: components/about.py
import subprocess
import os
from PyQt6.QtWidgets import QLabel, QWidget
from PyQt6.QtGui import QFont, QPixmap
from PyQt6.QtCore import Qt

ASSETS_PATH = os.path.expanduser("~/Elysia/assets/assets")
FASTFETCH_CMD = ["fastfetch", "-c", os.path.expanduser("~/.config/fastfetch/sys-config.jsonc")]


class AboutManager:
    def __init__(self, parent):
        self.parent = parent
        self.labels = []

        self.about_box = QWidget(self.parent)
        self.about_box.setGeometry(500, 100, 600, 640)
        self.about_box.setStyleSheet("""
            background-color: rgba(255, 255, 255, 0.05);
            border: 2px solid rgba(255, 255, 255, 0.15);
            border-radius: 30px;
        """)
        self.about_box.setVisible(False)

        # Desktop image
        self.image = QLabel(self.about_box)
        self.image.setPixmap(QPixmap(os.path.join(ASSETS_PATH, "desktop.png")).scaled(120, 120, Qt.AspectRatioMode.KeepAspectRatio, Qt.TransformationMode.SmoothTransformation))
        self.image.setGeometry(0, 0, 580, 170)
        self.image.setStyleSheet("color: white; background: transparent; border: none; margin: 0px; padding: 0px")
        self.image.setAlignment(Qt.AlignmentFlag.AlignCenter)

        # Title
        title = QLabel("SYSTEM INFORMATION", self.about_box)
        title.setFont(QFont("ElysiaOSNew", 20, QFont.Weight.Bold))
        title.setStyleSheet("color: white; background: transparent; border: none; margin: 0px; padding: 0px")
        title.setAlignment(Qt.AlignmentFlag.AlignCenter)
        title.setGeometry(155, 90, 290, 170)

        # Fastfetch output labels
        self.fastfetch_lines = self.get_fastfetch_output()
        y = 210
        for line in self.fastfetch_lines:
            label = QLabel(line.strip(), self.about_box)
            label.setFont(QFont("ElysiaOSNew22", 16))
            label.setStyleSheet("color: white; background: transparent; border: none; margin: 0px; padding: 0px")
            label.setGeometry(70, y, 480, 30)
            label.setWordWrap(True)
            label.setAlignment(Qt.AlignmentFlag.AlignLeft)
            label.setVisible(True)
            self.labels.append(label)
            y += 30

    def get_fastfetch_output(self):
        try:
            output = subprocess.check_output(FASTFETCH_CMD).decode()
            return [line for line in output.strip().splitlines() if line.strip()]
        except Exception as e:
            return [f"Error: {e}"]

    def show(self):
        self.parent.hide_tiles()
        self.about_box.setVisible(True)
        self.parent.show_back_button()

    def hide(self):
        self.about_box.setVisible(False)
        self.parent.hide_back_button()
        self.parent.show_tiles()

File: components/About.cpp
#include "About.h"
#include <QDir>
#include <QFont>
#include <QPixmap>
#include <QProcess>
#include <QWidget>
#include <QLabel>

About::About(QObject *parent) : QObject(parent) {
    QWidget *parentWidget = qobject_cast<QWidget*>(parent);
    if (!parentWidget) return;
    
    aboutBox = new QWidget(parentWidget);
    aboutBox->setGeometry(500, 100, 600, 640);
    aboutBox->setStyleSheet(
        "background-color: rgba(255, 255, 255, 0.05);"
        "border: 2px solid rgba(255, 255, 255, 0.15);"
        "border-radius: 30px;"
    );
    aboutBox->setVisible(false);
    
    // Desktop image
    image = new QLabel(aboutBox);
    QPixmap desktopPixmap(QDir::homePath() + "/.config/Elysia/assets/assets/desktop.png");
    image->setPixmap(desktopPixmap.scaled(120, 120, Qt::KeepAspectRatio, Qt::SmoothTransformation));
    image->setGeometry(0, 0, 580, 170);
    image->setStyleSheet("color: white; background: transparent; border: none; margin: 0px; padding: 0px");
    image->setAlignment(Qt::AlignCenter);
    
    // Title
    QLabel *title = new QLabel("SYSTEM INFORMATION", aboutBox);
    QFont titleFont("ElysiaOSNew", 20, QFont::Bold);
    title->setFont(titleFont);
    title->setStyleSheet("color: white; background: transparent; border: none; margin: 0px; padding: 0px");
    title->setAlignment(Qt::AlignCenter);
    title->setGeometry(155, 90, 290, 170);
    
    // Fastfetch output labels
    fastfetchLines = getFastfetchOutput();
    int y = 210;
    for (const QString &line : fastfetchLines) {
        QLabel *label = new QLabel(line.trimmed(), aboutBox);
        QFont font("ElysiaOSNew22", 16);
        label->setFont(font);
        label->setStyleSheet("color: white; background: transparent; border: none; margin: 0px; padding: 0px");
        label->setGeometry(70, y, 480, 30);
        label->setWordWrap(true);
        label->setAlignment(Qt::AlignLeft);
        label->setVisible(true);
        labels.append(label);
        y += 30;
    }
}

QStringList About::getFastfetchOutput() {
    QStringList result;
    QProcess process;
    QStringList args;
    args << "-c" << QDir::homePath() + "/.config/fastfetch/sys-config.jsonc";
    
    process.start("fastfetch", args);
    if (process.waitForFinished()) {
        QString output = QString::fromUtf8(process.readAllStandardOutput());
        QStringList lines = output.split('\n', Qt::SkipEmptyParts);
        for (const QString &line : lines) {
            if (!line.trimmed().isEmpty()) {
                result.append(line);
            }
        }
    } else {
        result.append("Error: Failed to run fastfetch");
    }
    
    return result;
}

void About::show() {
    QWidget *parentWidget = qobject_cast<QWidget*>(parent());
    if (parentWidget) {
        // Hide tiles and show back button (would need to be implemented in MainWindow)
        aboutBox->setVisible(true);
    }
}

void About::hide() {
    aboutBox->setVisible(false);
    // Show tiles and hide back button (would need to be implemented in MainWindow)
} 